lib/SILGen/SILGenExpr.cpp

```diff
@@ -3782,7 +3783,48 @@ visitMagicIdentifierLiteralExpr(MagicIdentifierLiteralExpr *E, SGFContext C) {
   llvm_unreachable("Unhandled MagicIdentifierLiteralExpr in switch.");
 }
 
-RValue RValueEmitter::visitCollectionExpr(CollectionExpr *E, SGFContext C) {
+RValue RValueEmitter::visitArrayExpr(ArrayExpr *E, SGFContext C) {
+  auto loc = SILLocation(E);
+  auto init = E->getInitializer();
+  auto *decl = dyn_cast<AbstractFunctionDecl>(init.getDecl());
+  Type elementType = E->getElementType();
+  ArrayRef<Identifier> argLabels = decl->getFullName().getArgumentNames();
+  CanType arrayTy = ArraySliceType::get(elementType)->getCanonicalType();
+  VarargsInfo varargsInfo =
+      emitBeginVarargs(SGF, loc, elementType->getCanonicalType(), arrayTy,
+                       E->getNumElements(), {});
+
+  for (unsigned index : range(E->getNumElements())) {
+    auto destAddr = varargsInfo.getBaseAddress();
+    if (index != 0) {
+      SILValue indexValue = SGF.B.createIntegerLiteral(
+          loc, SILType::getBuiltinWordType(SGF.getASTContext()), index);
+      destAddr = SGF.B.createIndexAddr(loc, destAddr, indexValue);
+    }
+    auto &destTL = varargsInfo.getBaseTypeLowering();
+    TemporaryInitialization init(destAddr, CleanupHandle::invalid());
+    ArgumentSource(E->getElements()[index])
+        .forwardInto(SGF, varargsInfo.getBaseAbstractionPattern(), &init,
+                     destTL);
+  }
+
+  RValue arg(SGF, loc, arrayTy,
+             emitEndVarargs(SGF, loc, std::move(varargsInfo)));
+
+  // Add an argument label.
+  assert(argLabels.size() == 1 && !argLabels[0].empty() &&
```

Comment:
Can you add a comment here explaining why you are doing this?

Reply:
(i.e. I have a guess on why you are doing this here... I shouldn't have to guess/thin/worry here. The comment will assuage those issues.

Reply:
This is because I'm calling a function of the form: init(arrayLiteral: T...), but I tested it without it, and this relabeling is not strictly necessary.

Reply:
If it is not necessary, lets remove it.

Reply:
The "relabeling" will go away with some refactoring I'm planning on doing in this area later. You don't have to worry about it for now.

Reply:
Whoop! missed this comment.